### PR TITLE
[stable8] Fix LDAP wizard saveMultiSelect race condition

### PR DESCRIPTION
**Pull request by PVince81**
Whenever a checkbox is ticked in a multiselect box, it will trigger the
saving and retrieving ajax calls at the same time.

This fix makes sure that it will first save, and only once saving is
done, do the call that retrieves the filter.

Fixes the issue that @davitol found while testing something else: https://github.com/owncloud/core/pull/16456#issuecomment-105870200

@davitol @MorrisJobke @blizzz

### CODE REVIEW

**Comment by PVince81** [comment]
I had a quick debugging session on master in the new wizard and saw that saving the configuration will properly wait before updating fields, which is the correct behavior. This means the problem will not appear on master.

This fix is specific for OC 8.0.x

File: apps/user_ldap/js/settings.js [hunk 1]
```diff
@@ -888,10 +902,12 @@ var LdapWizard = {
 				'&action=save'+
 				'&ldap_serverconfig_chooser='+$('#ldap_serverconfig_chooser').val();
 
-		$.post(
+
+		return $.post(
 			OC.filePath('user_ldap','ajax','wizard.php'),
 			param,
 			function(result) {
+				console.log('after save');
```
Comment (MorrisJobke) [inline apps/user_ldap/js/settings.js]: remove this?

Comment (PVince81) [inline apps/user_ldap/js/settings.js]: no after shave ? :wink: 
Missed it

**Comment by blizzz** [comment]
Better, but I still had some "out of sync" between (un)selected items and the filter. Changing the behaviour from save on click in the multiselect to save on close (as it is already on master) makes it work finally for me. With the disadvantage that save and count is done on every close, even if not change was done (not so in master, there is a different check, but this we cannot backport, at least not the way it was done). But better this than odd wizard behaviour.

**Comment by blizzz** [comment]
@davitol @jvillafanez mind testing?

**Comment by jvillafanez** [comment]
Are we allowed to do evil things? :innocent: 

In the group filter, "only from those groups", I've click it several times very fast and after a bit that select stopped working: you can choose and change the selection, but the request to save isn't being done. I'm not sure if this will happen with a slow network connection.

Other than that, I think it's working fine and better than before.

**Comment by blizzz** [comment]
> Are we allowed to do evil things? :innocent: 

Well… how else would you know what evil people would do? :smiling_imp: 

> In the group filter, "only from those groups", I've click it several times very fast [..]

I assume this is no common user behaviour :) For OC 8, we can live with it, more interesting for me is whether you see it with master/upcoming 8.1 where the wizard is sane ;)

> Other than that, I think it's working fine and better than before.

I take it as :+1: 

Another reviewer/tester? maybe @Xenopathic @tkaufm

**Comment by tkaufm** [comment]
@blizzz
will try to test it within the next few days, but cant promise

**Comment by davitol** [comment]
#16456 is still been reproduced

**Comment by MorrisJobke** [comment]
I will review this one.

**Comment by MorrisJobke** [comment]
It will now be refreshed on every click on the button to show the raw filter. This is far better than before.

For stable8 this is enough I think :+1:

cc @davitol

**Comment by tkaufm** [comment]
@blizzz

Just checked an it worked for me.

But I realized I always gave it a direct ldap-filter. I started with OC5 and from there it was updated without any problems. The only thing i did in OC7 was to select the 'Manually enter LDAP filter' and reenter the filters again.

Hope it helps.

Best
Tobias

**Comment by DeepDiver1975** [comment]
@jnfrmarks  @davitol please retest - thx

**Comment by davitol** [comment]
as @MorrisJobke said "It will now be refreshed on every click on the button to show the raw filter.", so if it is good enough :+1:

**Comment by blizzz** [comment]
thx @tkaufm and @davitol

**Pull request merged**